Source/WebKit/UIProcess/skia/BackingStoreSkia.cpp

```diff
@@ -29,15 +29,39 @@
 #if USE(SKIA)
 
 #include "UpdateInfo.h"
-#include <WebCore/NotImplemented.h>
+#include <WebCore/GraphicsContextSkia.h>
+#include <skia/core/SkBitmap.h>
+#include <skia/core/SkCanvas.h>
 
 namespace WebKit {
 
 static const Seconds s_scrollHysteresisDuration { 300_ms };
 
+static sk_sp<SkSurface> createSkiaImageSurfaceWithFastMalloc(const WebCore::IntSize& size, double deviceScaleFactor)
+{
+    ASSERT(!size.isEmpty());
+    WebCore::IntSize scaledSize(size);
+    scaledSize.scale(deviceScaleFactor);
+    auto info = SkImageInfo::MakeN32Premul(scaledSize.width(), scaledSize.height(), SkColorSpace::MakeSRGB());
+    auto* pixels = fastZeroedMalloc(scaledSize.height() * info.minRowBytes());
+    return SkSurfaces::WrapPixels(info, pixels, info.minRowBytes(), nullptr);
+}
+
+static void copyRectFromOneSurfaceToAnother(SkSurface* source, SkSurface* destination, const WebCore::IntSize& sourceOffset, const WebCore::IntRect& rect)
+{
+    SkBitmap bitmap;
+    bitmap.allocPixels(source->imageInfo().makeWH(rect.width(), rect.height()));
+
+    if (!source->readPixels(bitmap, sourceOffset.width(), sourceOffset.height()))
+        return;
+
+    destination->writePixels(bitmap, rect.x(), rect.y());
+}
```

Comment:
Think I translated from Cairo wrong with this one but getting closer. @carlosgcampos not sure if you have any thoughts on how this should be implemented here.